src/typechecker.rs

```diff
@@ -4766,6 +4766,14 @@ pub fn typecheck_expression(
                 }
             }
 
+            let parent_throws = project.get_scope(scope_id).throws;
+            if fill_size_expr.is_some() && !parent_throws {
+                error = error.or(Some(JaktError::TypecheckError(
+                    "Prefilling an array may throw and needs to be in a try statement or a function marked as throws".to_string(),
+                    fill_size_expr.as_ref().unwrap().span(),
```

Comment:
if you move this lower, inside of the `if let Some(fill_size_expr) ...` you can simplify and not have to do the `is_some()` or the `.as_ref().unwrap()` step. Should make it a little easier.